clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2837,9 +2837,9 @@ def CIR_InlineAsmOp : CIR_Op<"asm", [RecursiveMemoryEffects]> {
     ... 
     %2 = cir.load %0 : cir.ptr <!s32i>, !s32i
     %3 = cir.load %1 : cir.ptr <!s32i>, !s32i
-    cir.asm(x86_att, {"foo" "~{dirflag},~{fpsr},~{flags}"} : () -> ()
-    cir.asm(x86_att, {"bar $$42 $0" "=r,=&r,1,~{dirflag},~{fpsr},~{flags}"} %2 : (!s32i) -> ()
-    cir.asm(x86_att, {"baz $$42 $0" "=r,=&r,0,1,~{dirflag},~{fpsr},~{flags}"} %3, %2 : (!s32i, !s32i) -> ()
+    cir.asm(has_side_effects, x86_att, {"foo" "~{dirflag},~{fpsr},~{flags}"}) : () -> ()
```

Comment:
Can you move this to after the enclosing param and rename it a bit? Example: `cir.asm(x86_att, {"foo" "~{dirflag},~{fpsr},~{flags}"}) side_effects